HPE MCP repository: Fix jessie/stretch packages, allow setting repository version

matsimon opened the pull request:
This should address issues as found in !2

**Fix package names for Debian 8/9**
Add the same packages as are used in RedHat.yml for Debian 8 and 9 where they do exist but not 7. 
* hpacucli isn't present in MCP repository releases anything (likely newer than 10.40), see details at HPE about their [Management Component Pack](http://downloads.linux.hpe.com/SDR/project/mcp/).
* hpssacli was renamed to ssacli

**Introduce a variable that allows overriding the MCP repository version**
Defaulting to current (latest) MCP version is sane for most cases, but there are actual known cases where we do want to specify a certain release because of compatibility issues with either hard- or software.

Commit:
Debian: Fix packages present in HPE repositories for jessie/stretch

The current MCP repositories for both jessie and stretch dropped
hpacucli and hpssacli for ssacli.

Depending on HPE's way to go forward, we may have to adapt again since
the 'current' MCP repository doesn't equal the same version across
distribution versions.

Partially addresses !2

Commit:
HPE repository: Define version of the MCP repository

Default to current, but lets you override it, if hardware needs specific
versions (or versions in the current repo have known issues).

Commit:
RedHat: Alphabetical ordering of packages

matsimon commented:
I'd like to document the introduced variables, any suggestions @keachi ?

keachi commented on defaults/main.yml:
```diff
@@ -1 +1,4 @@
 ---
+
+hwraid_hp_repository_version: 'current'
+
```
Remove the empty line at the end of the file.

keachi commented on vars/Debian.yml:
```diff
@@ -10,15 +10,16 @@ hwraid_apt_le_vert_keys:
 
 # HP apt repository
-hwraid_apt_hp_repository: 'deb http://downloads.linux.hpe.com/SDR/repo/mcp/{{ ansible_distribution | lower }} {{ ansible_distribution_release | lower }}/current non-free'
+hwraid_apt_hp_repository: 'deb http://downloads.linux.hpe.com/SDR/repo/mcp/{{ ansible_distribution | lower }} {{ ansible_distribution_release | lower }}/{{ hwraid_hp_repository_version }}'
```
You define `hwraid_hp_repository_version: 'current'`, but you change `current non-free` with that variable. Is that your idea and `non-free` isn't used any longer?

matsimon replied:
Of course "non-free" needs to stay, I've amended that error with 176e5e3194b99941945175eb74465df8c521ca3a

keachi commented on vars/Debian.yml:
```diff
@@ -10,15 +10,16 @@ hwraid_apt_le_vert_keys:
-  - hpssacli
+  - ssa
+  - ssacli
+  - ssaducli
```
Do we need those packages in every case or do we need sometimes other packages depending on hardware raid controller? If the case is that we have to change the packages, those packages should be defined here in a variable and in the `defaults/main.yml` should be a intermediate variable which is later used in the task.

task -> defaults -> vars (depending on OS)

matsimon replied:
You did define them in RedHat.yml so I chose to allign with what was defined there ;)

* ssacli being a (almost) direct replacement for hpacucli should be present on any system with a HPE RAID conterollers.
* ssaducli looks pretty sane to me as well since it is the diagnostics utility for them.
* ssa is the one I'm a bit reluctant to say yes. I've both never used it, nor have I found yet the thing making it essential to us.

So far, if a HP(E) volume is detected we can mostly cound on HPE to support them with these tools if they support it by providing required packages for the required OS/HW combination. (unlike megacli/storcli/sas[2|3]ircu from LSI/Broadcom)

I don't think we can count on HPE to keep package names between Debian-based and EL distribution identical. We can likely assume the package names are the same for Debian distributions or EL distributions, but not both using the same package naming.

keachi commented on vars/RedHat.yml:
```diff
@@ -5,16 +5,16 @@ hwraid_yum_hp_repository:
-  - ssacli
   - ssa
+  - ssacli
+  - ssaducli
```
Same here as for debian based systems.

Commit:
vars/Debian: Fix repository definition

Of course we need to keep it and since HPE only provides a
section "non-free", this value needs to both stay there as is
and not be changeable for the time being.

keachi reviewed (approved):
LGTM
